src/schema/fields/pagination.js

```diff
@@ -0,0 +1,48 @@
+import {
+  GraphQLInt,
+  GraphQLObjectType,
+  GraphQLString,
+  GraphQLNonNull,
+} from "graphql"
+import { toGlobalId } from "graphql-relay"
+
+export const PageCursorsType = new GraphQLObjectType({
+  name: "PageCursors",
+  fields: () => ({
+    first: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "Cursor that starts the first page",
+    },
+    last: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "Cursor that starts the last page",
+    },
+    next: {
+      type: GraphQLString,
+      description: "If there is a next page, cursor that starts it.",
+    },
+    prev: {
+      type: GraphQLString,
+      description: "If there is a previous page, cursor that starts it.",
+    },
+    currentPage: {
+      type: new GraphQLNonNull(GraphQLInt),
+    },
+    totalPages: {
+      type: new GraphQLNonNull(GraphQLInt),
+    },
+  }),
+})
+
+const PREFIX = "arrayconnection"
+export function createPageCursors({ page, size }, total) {
+  const totalPages = Math.ceil(total / size)
+  return {
```

Comment:
There's probably an 'off-by-1' somewhere here..this was just quick to get a prototype up (and work on the Reaction component to use this). I'll circle back if we go this route with proper specs, etc.

Reply:
Yeah, I tend to have things be zero - based and name them `*Index` to try to indicate that, which often makes the math more representative of the logic, since there are few -1's.